File: src/linkedlist/Solution206.java
package linkedlist;// 206. 反转链表


public class Solution206 {

    public ListNode reverseList(ListNode head) {

        ListNode last = null;
        ListNode now = head;

        while (now != null) {
            ListNode tmp = now.next;
            now.next = last;
            last = now;
            now = tmp;
        }

        return last;

    }

}

File: python/linked_list/Solution206.py
# 206. 反转链表
from typing import Optional

from linked_list import ListNode


class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        last = None
        now = head
        while now is not None:
            tmp = now.next
            now.next = last
            last = now
            now = tmp
        return last

